pyproject.toml

```diff
@@ -94,8 +94,9 @@ replacement = 'src="https://raw.githubusercontent.com/encode/httpx/master/\1"'
 # https://beta.ruff.rs/docs/configuration/#using-rufftoml
 [tool.ruff]
 select = ["E", "F", "I", "B", "PIE"]
-ignore = ["B904", "B028"]
-line-length = 120
```

Comment:
Previously, we had a 120 limit on the ruff linter and an 88 limit on the black formatter, so our entire code base had an 88 limit.

Reply:
Not technically true, as Black's limit is used to attempt to split lines but sometimes it cannot. Ruff would raise E501 on lines exceeding 120 which was the "true" line limit. Now you've disabled E501 so lines can exceed the maximum length without warning.

Personally I think having line-length enforcement in addition to the formatter is overkill, but this is a change.

Reply:
I believe we should keep the real line-length limit at 120, as we did previously, and not ignore E501 so ruff can warn us about long strings that ruff format cannot fix.

Reply:
I think for now you can keep current behavior by this configuration:

```toml
# https://docs.astral.sh/ruff/configuration/#using-rufftoml
[tool.ruff]
select = ["E", "F", "I", "B", "PIE"]
ignore = ["B904", "B028"]

[tool.ruff.pycodestyle]
max-line-length = 120

[tool.ruff.isort]
combine-as-imports = true
```
Ref: https://docs.astral.sh/ruff/settings/#pycodestyle-max-line-length
Use default `line-length` for formatting and warn about long strings via `max-line-length`.

Reply:
> I think for now you can keep current behavior by this configuration

That seems helpful, thanks.

Their announcement includes this...

> "ruff format achieves >99.9% compatibility with Black, as measured by changed lines. When migrating an existing project to Ruff’s formatter, the vast majority of lines will be formatted identically"

We ought to aim for this PR having as minimal a change footprint as possible.

Reply:
> I'm not sure why the formatter limit is 88 and the linter limit is 120.

The obvious guess is the formatter is an ecosystem replacement to black, and the linter is a replacement for flake8.

We aim to have the same setup/configuration/organization across the projects of encode, and the decision here will affect the others.

Also, we always have a policy of minimal changes on each PR.

Reply:
> We aim to have the same setup/configuration/organization across the projects of encode, and the decision here will affect the others.

Yes, it makes sense now why we might want to continue using this approach. 
Compared to the request to "change back to 88," this explanation is far more detailed.

Anyway, if you like, this pull request could be a good starting point for migrating to the "same limitation for linter and formatter" approach across the encode projects.

Reply:
> Also, we always have a policy of minimal changes on each PR.

👍 - I'm okay with reviewing this pr if/when it doesn't substantially affect the existing formatting.

Reply:
> Anyway, if you like, this pull request could be a good starting point for migrating to the "same limitation for linter and formatter" approach across the encode projects

As you also pointed there are some cases (e.g. long strings) formatter unable to keep line length so as POV of formatter the code is formatted, but linter will raise error on that line.
Here are all lines raises `E501` with line length > 88:
```shell
> ruff check .
httpx\_client.py:1267:89: E501 Line too long (96 > 88)
httpx\_client.py:1983:89: E501 Line too long (96 > 88)
httpx\_decoders.py:262:89: E501 Line too long (89 > 88)
httpx\_exceptions.py:316:89: E501 Line too long (99 > 88)
httpx\_exceptions.py:326:89: E501 Line too long (98 > 88)
httpx\_main.py:66:89: E501 Line too long (95 > 88)
httpx\_main.py:78:89: E501 Line too long (92 > 88)
httpx\_models.py:361:89: E501 Line too long (93 > 88)
httpx\_multipart.py:51:89: E501 Line too long (96 > 88)
httpx\_multipart.py:99:89: E501 Line too long (94 > 88)
httpx\_multipart.py:103:89: E501 Line too long (93 > 88)
httpx\_multipart.py:121:89: E501 Line too long (94 > 88)
httpx\_transports\default.py:193:89: E501 Line too long (108 > 88)
httpx\_transports\default.py:331:89: E501 Line too long (108 > 88)
httpx\_urlparse.py:363:89: E501 Line too long (93 > 88)
httpx\_urlparse.py:379:89: E501 Line too long (97 > 88)
httpx\_urlparse.py:458:89: E501 Line too long (89 > 88)
httpx\_urls.py:54:89: E501 Line too long (91 > 88)
httpx\_urls.py:60:89: E501 Line too long (89 > 88)
httpx\_urls.py:67:89: E501 Line too long (89 > 88)
httpx\_urls.py:118:89: E501 Line too long (93 > 88)
httpx\_urls.py:308:89: E501 Line too long (92 > 88)
httpx\_urls.py:345:89: E501 Line too long (106 > 88)
tests\client\test_auth.py:599:89: E501 Line too long (94 > 88)
tests\models\test_cookies.py:97:89: E501 Line too long (105 > 88)
tests\test_config.py:104:89: E501 Line too long (98 > 88)
tests\test_multipart.py:177:89: E501 Line too long (106 > 88)
tests\test_utils.py:112:89: E501 Line too long (100 > 88)
Found 28 errors.
```

You can fix them in separate PR then remove `max-line-length = 120`.